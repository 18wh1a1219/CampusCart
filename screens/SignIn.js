import React, { useState } from 'react';
import { View, ImageBackground, Text, TextInput, StyleSheet, Alert } from 'react-native';
import { TouchableOpacity } from 'react-native-gesture-handler';
import { Image } from 'react-native-web';
import {signIn} from '../API/firebaseMethods';
import { FloatingLabelInput } from './FloatingLabelInput';

export default function SignIn({ navigation }) {
  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');

  const handlePress = () => {
    if (!email) {
      Alert.alert('Email field is required.');
    }

    if (!password) {
      Alert.alert('Password field is required.');
    }

    signIn(email, password);
    setEmail('');
    setPassword('');
  };

  return (
    <ImageBackground
      style={styles.background}
      source={require('../assets/Signing.png')}>
      <Text style={styles.text}>Sign In </Text>
      <View style={styles.in} >
      <FloatingLabelInput
      
        label="Email"
        value={email}
        onChangeText={(email) => setEmail(email)}
        autoCapitalize="none"
      />
      <FloatingLabelInput
       
         label="Password"
        value={password}
        onChangeText={(password) => setPassword(password)}
        secureTextEntry={true}
      />
</View>
      <TouchableOpacity style={styles.button} onPress={handlePress}>
        <Text style={styles.buttonText}>Sign In</Text>
      </TouchableOpacity>

    </ImageBackground>
  );
}
const styles = StyleSheet.create({
  background: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    height:'20%'
  },
  button: {
    width: 350,
    height:'25%',
    borderRadius: 15,
    borderWidth: 3,
    borderColor: '#ffd60a',
    backgroundColor: '#ffd60a',
    color: 'white',
    margin: '2%',
    alignSelf:'center',
    padding:'2%'

  },
  buttonText: {
    fontSize:18,
    color: 'black',
    fontWeight: 'bold',
    textAlign: 'center',
  },
  container: {
    height: '100%',
    width: '100%',
    backgroundColor: '#ECF3FB',
    alignItems: 'center',
    justifyContent: 'center',
  },
 
  text: {
    textAlign: 'center',
    fontSize: 22,
    margin: 10,
    fontWeight: 'bold',
    color: 'black',
    marginTop:90
  },
  in:{
    marginBottom:25,
    marginTop:25
  }
});