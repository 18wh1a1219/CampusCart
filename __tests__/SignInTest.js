import React from 'react';
import renderer from 'react-test-renderer';
import SignIn from '../screens/SignIn';

test('renders correctly', () => {
  const tree = renderer.create(<SignIn />).toJSON();
  expect(tree).toMatchSnapshot();
});
