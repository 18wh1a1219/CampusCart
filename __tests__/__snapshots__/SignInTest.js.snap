// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "height": "20%",
      "justifyContent": "center",
    }
  }
>
  <Image
    source={
      Object {
        "process": [Function],
      }
    }
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": "20%",
          "width": undefined,
        },
        undefined,
      ]
    }
  />
  <Text
    style={
      Object {
        "color": "black",
        "fontSize": 22,
        "fontWeight": "bold",
        "margin": 10,
        "marginTop": 90,
        "textAlign": "center",
      }
    }
  >
    Sign In 
  </Text>
  <View
    style={
      Object {
        "marginBottom": 25,
        "marginTop": 25,
      }
    }
  >
    <View
      style={
        Object {
          "paddingTop": 18,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#aaa",
            "fontSize": 15,
            "left": 0,
            "position": "absolute",
            "top": 18,
          }
        }
      >
        Email
      </Text>
      <TextInput
        autoCapitalize="none"
        blurOnSubmit={true}
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        style={
          Object {
            "borderBottomColor": "#555",
            "borderBottomWidth": 1,
            "color": "#000",
            "fontSize": 20,
            "height": 50,
            "marginBottom": 15,
            "width": 350,
          }
        }
        value=""
      />
    </View>
    <View
      style={
        Object {
          "paddingTop": 18,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#aaa",
            "fontSize": 15,
            "left": 0,
            "position": "absolute",
            "top": 18,
          }
        }
      >
        Password
      </Text>
      <TextInput
        blurOnSubmit={true}
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        secureTextEntry={true}
        style={
          Object {
            "borderBottomColor": "#555",
            "borderBottomWidth": 1,
            "color": "#000",
            "fontSize": 20,
            "height": 50,
            "marginBottom": 15,
            "width": 350,
          }
        }
        value=""
      />
    </View>
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignSelf": "center",
        "backgroundColor": "#ffd60a",
        "borderColor": "#ffd60a",
        "borderRadius": 15,
        "borderWidth": 3,
        "color": "white",
        "height": 73,
        "margin": "2%",
        "opacity": 1,
        "padding": "5%",
        "width": 350,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "black",
          "fontSize": 18,
          "fontWeight": "bold",
          "textAlign": "center",
        }
      }
    >
      Sign In
    </Text>
  </View>
</View>
`;
